threedi_modelchecker/config.py

```diff
@@ -1176,221 +1142,215 @@ def is_none_or_empty(col):
-            models.GlobalSetting.maximum_table_step_size
-            < models.GlobalSetting.table_step_size,
+        column=models.ModelSettings.maximum_table_step_size,
+        invalid=Query(models.ModelSettings).filter(
+            models.ModelSettings.maximum_table_step_size
+            < models.ModelSettings.minimum_table_step_size,
         ),
-        message="v2_global_settings.maximum_table_step_size should be greater than v2_global_settings.table_step_size.",
+        message="model_settings.maximum_table_step_size should be greater than model_settings.minimum_table_step_size.",
     ),
     QueryCheck(
         error_code=325,
         level=CheckLevel.WARNING,
-        column=models.GlobalSetting.interception_global,
-        invalid=Query(models.GlobalSetting).filter(
-            first_setting_filter,
-            ~is_none_or_empty(models.GlobalSetting.interception_file),
-            is_none_or_empty(models.GlobalSetting.interception_global),
+        column=models.Interception.interception,
+        invalid=Query(models.Interception).filter(
+            ~is_none_or_empty(models.Interception.interception_file),
+            is_none_or_empty(models.Interception.interception),
         ),
-        message="v2_global_settings.interception_global is recommended as fallback value when using an interception_file.",
+        message="interception.interception is recommended as fallback value when using an interception_file.",
     ),
 ]
 
 CHECKS += [
-    QueryCheck(
-        error_code=326,
-        level=CheckLevel.INFO,
-        column=table.id,
-        invalid=Query(table).filter(
-            table.id != Query(setting).filter(first_setting_filter).scalar_subquery()
-        ),
-        message=f"{table.__tablename__} is defined, but not referred to in v2_global_settings.{setting.name}",
+    UsedSettingsPresentCheck(
+        error_code=326, level=CheckLevel.ERROR, column=use_col, settings_table=table
     )
-    for table, setting in (
+    for table, use_col in (
         (
             models.SimpleInfiltration,
-            models.GlobalSetting.simple_infiltration_settings_id,
+            models.ModelSettings.use_simple_infiltration,
         ),
-        (models.Interflow, models.GlobalSetting.interflow_settings_id),
-        (models.GroundWater, models.GlobalSetting.groundwater_settings_id),
-        (models.NumericalSettings, models.GlobalSetting.numerical_settings_id),
-        (models.ControlGroup, models.GlobalSetting.control_group_id),
-        (models.VegetationDrag, models.GlobalSetting.vegetation_drag_settings_id),
+        (models.Interflow, models.ModelSettings.use_interflow),
+        (models.GroundWater, models.ModelSettings.use_groundwater_flow),
+        (models.GroundWater, models.ModelSettings.use_groundwater_storage),
+        (models.VegetationDrag, models.ModelSettings.use_vegetation_drag_2d),
+        (models.Interception, models.ModelSettings.use_interception),
     )
 ]
```

Comment:
Check replaced with custom check that checks if a row is present if `use_settings = True`. A custom check is used because easy joining is not possible without foreign key.